common/test/acceptance/tests/lms/test_account_settings.py

```diff
@@ -266,28 +273,21 @@ def test_email_field(self):
             assert_after_reload=False
         )
 
-        self.verify_events_of_type(
-            username,
-            self.CHANGE_INITIATED_EVENT_NAME,
+        actual_events = self.wait_for_events(
+            event_filter=self.settings_change_initiated_event_filter, number_of_matches=2)
```

Comment:
One would have to wait to see if extra events showed up, rather than only waiting for the specified number of matches.  That would require setting an appropriate timeout -- not too long, not too short, that would be applied *after* the specified matches are made.  It could be a "strict" mode, so it could be disabled in cases where extra events are okay.